File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        count = 0
        product = 1
        n = len(nums)
        end = 0
        for start in range(n):
            product *= nums[start]
            while(product >= k and start >= end):
                product /= nums[end]
                end+= 1
            count += (start - end + 1)
        return count

        

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int count = 0 ;
        int product = 1;
        int end = 0;
        for(int start = 0 ; start < nums.size() ; start++){
            product *= nums[start];
            while(product >= k and start >= end){
                product /= nums[end];
                end++;
                
            }
            count += (start - end + 1);
        }
        return count;
        }
};